src/web/app/pages-admin/admin-notifications-page/admin-notifications-page.component.spec.ts

```diff
@@ -1,30 +1,288 @@
 import { HttpClientTestingModule } from '@angular/common/http/testing';
 import { ComponentFixture, TestBed } from '@angular/core/testing';
-
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import moment from 'moment-timezone';
+import { of, throwError } from 'rxjs';
+import SpyInstance = jest.SpyInstance;
+import { NotificationService } from '../../../services/notification.service';
+import { SimpleModalService } from '../../../services/simple-modal.service';
+import { StatusMessageService } from '../../../services/status-message.service';
+import { TimezoneService } from '../../../services/timezone.service';
+import { createMockNgbModalRef } from '../../../test-helpers/mock-ngb-modal-ref';
+import { Notification, NotificationStyle, NotificationTargetUser } from '../../../types/api-output';
+import { SortBy } from '../../../types/sort-properties';
+import { SimpleModalType } from '../../components/simple-modal/simple-modal-type';
 import { AdminNotificationsPageComponent } from './admin-notifications-page.component';
 import { AdminNotificationsPageModule } from './admin-notifications-page.module';
+import { NotificationEditFormModel } from './notification-edit-form/notification-edit-form-model';
+import { NotificationsTableRowModel } from './notifications-table/notifications-table-model';
+
+const exampleNotificationEditModel: NotificationEditFormModel = {
+  notificationId: 'notification1',
+  shown: false,
+
+  startTime: { hour: 0, minute: 0 },
+  startDate: { year: 0, month: 0, day: 0 },
+  endTime: { hour: 0, minute: 0 },
+  endDate: { year: 0, month: 0, day: 0 },
+
+  style: NotificationStyle.SUCCESS,
+  targetUser: NotificationTargetUser.INSTRUCTOR,
+
+  title: 'valid title',
+  message: 'valid message',
+
+  isSaving: false,
+  isDeleting: false,
+};
+
+const exampleNotificationOne: Notification = {
+  notificationId: 'notification1',
+  startTimestamp: moment('2017-09-15 09:30:00').valueOf(),
+  endTimestamp: moment('2050-09-15 09:30:00').valueOf(),
+  createdAt: moment('2017-09-15 09:30:00').valueOf(),
+  updatedAt: moment('2017-09-15 09:30:00').valueOf(),
+  style: NotificationStyle.SUCCESS,
+  targetUser: NotificationTargetUser.INSTRUCTOR,
+  title: 'valid title 1',
+  message: 'valid message 1',
+  shown: false,
+};
+
+const exampleNotificationTwo: Notification = {
+  notificationId: 'notification2',
```

Comment:
```suggestion

const testNotification2: Notification = {
  notificationId: 'notification2',
```

Reply:
Okay sure, thank you for your suggestions!